File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/_deprecated/IntakeExtendAndReady.java
package org.firstinspires.ftc.teamcode._deprecated;

import com.arcrobotics.ftclib.command.CommandBase;

import org.firstinspires.ftc.teamcode.state.SubsystemStates;

// intake-only, does full intake stuff until transfer
public class IntakeExtendAndReady extends CommandBase {
    MyRobot robot;
    Intake intake;

    public IntakeExtendAndReady(Intake intake, MyRobot robot) {
        this.intake = intake;
        this.robot = robot;

        addRequirements(intake);
    }

    @Override
    public void initialize() {
        intake.intakeBusy = true;
        intake.intakeOff();
    }

    public void execute() {
        switch (intake.state) {
            case IDLE:
                intake.setState(SubsystemStates.IntakeStates.ACTIVATED);
                break;
            case ACTIVATED:
                intake.setState(SubsystemStates.IntakeStates.EXTENDING);
                break;
              case EXTENDING:
                intake.extendSlides();
                intake.setState(SubsystemStates.IntakeStates.EXTENDED);
                break;
        }
    }

    @Override
    public boolean isFinished() {
        if(intake.state == SubsystemStates.IntakeStates.EXTENDED) {
            intake.intakeBusy = false;
            return true;
        }
        return false;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/commands/IntakeExtendAndReady.kt
package org.firstinspires.ftc.teamcode.commands

import com.arcrobotics.ftclib.command.CommandBase
import org.firstinspires.ftc.teamcode.core.Robot
import org.firstinspires.ftc.teamcode.core.SubsystemStates.*
import org.firstinspires.ftc.teamcode.subsystems.Intake
import org.firstinspires.ftc.teamcode.core.Constants.*

class IntakeExtendAndReady(val robot: Robot, val intake: Intake) : CommandBase() {
    init {
        addRequirements(intake)
    }

    override fun initialize() {
        intake.intakeBusy = false
        intake.intakeOff()
    }

    override fun execute() {
        when (intake.state) {
            IntakeStates.IDLE -> intake.state = IntakeStates.ACTIVATED
            IntakeStates.ACTIVATED -> intake.state = IntakeStates.EXTENDING
            IntakeStates.EXTENDING -> {
                intake.extendSlides()
                if (intake.slidesAtTarget()) intake.state = IntakeStates.EXTENDED
            }
            IntakeStates.EXTENDED -> {
                intake.wristToPos(INTAKE_WRIST_DOWN)
                intake.state = IntakeStates.EXTENDED_AND_DOWN
            }
            else -> {}
        }
    }

    override fun isFinished(): Boolean {
        if (intake.state == IntakeStates.EXTENDED_AND_DOWN) {
            intake.intakeBusy = false
            return true
        }
        return false
    }
}